### PR TITLE
General media upload

### LINKED ISSUE #412: Advanced photo uploads
~For the beta,~ photos should be able to be uploaded for the following:

- Profile pic
- Me
- My home/couch
- Hosting/surfing photos
- Travel photos

Or alternatively, just my home/couch and custom albums. Photos should be able to be ordered, deleted, and have captions (associated text).

### PR DESCRIPTION
**Pull request by aapeliv**
Closes #412 

TODO:

- [x] Convert existing uploads into new system, people don't want to re-upload their profile pics
- [x] Modify profile update RPC to set avatar filename
- [x] Add database columns to other objects that should have images
- [x] Update RPCs to set the images for other objects that can have pictures
- [ ] ~~Albums, etc~~ not for beta

### CODE REVIEW

File: app/pb/pages.proto [hunk 1]
```diff
@@ -69,13 +70,15 @@ message Page {
 message CreatePlaceReq {
   string title = 1;
   string content = 2;
+  string photo_key = 5;
```
Comment (kalvdans) [inline app/pb/pages.proto:76]: // from media server

File: app/pb/pages.proto [hunk 2]
```diff
@@ -69,13 +70,15 @@ message Page {
...
   string address = 3;
   Coordinate location = 4;
 }
 
 message CreateGuideReq {
   string title = 1;
   string content = 2;
+  string photo_key = 6;
```
Comment (kalvdans) [inline app/pb/pages.proto:85]: // from media server

File: app/pb/pages.proto [hunk 3]
```diff
@@ -90,6 +93,8 @@ message UpdatePageReq {
 
   google.protobuf.StringValue title = 2;
   google.protobuf.StringValue content = 3;
+  // set to empty string to clear
```
Comment (kalvdans) [inline app/pb/pages.proto]: // from media server, set to empty string to clear

File: app/pb/pages.proto [hunk 4]
```diff
@@ -60,6 +60,7 @@ message Page {
   int64 thread_id = 10;
   string title = 11;
   string content = 12;
+  string photo_url = 19;
```
Comment (kalvdans) [inline app/pb/pages.proto:64]: // empty string if no photo

File: app/backend/src/tests/test_email.py [hunk 5]
```diff
@@ -142,8 +147,8 @@ def mock_queue_email(sender_name, sender_email, recipient, subject, plain, html)
             assert from_date in html
             assert to_date in plain
             assert to_date in html
-            assert from_user.avatar_url not in plain
-            assert from_user.avatar_url in html
+            assert from_user.avatar_filename not in plain
+            assert from_user.avatar_filename in html
```
Comment (kalvdans) [inline app/backend/src/tests/test_email.py]: Hmm, didn't you just remove avatar_filename from the User/users table?

**Review by kalvdans — CHANGES_REQUESTED** [review]
Some small remarks only

File: app/backend/src/tests/test_email.py (hunk 5 above)
Comment (aapeliv) [reply app/backend/src/tests/test_email.py]: Yes, those tests are awful and crappy. It seems the code's not actually being run...

Comment (kalvdans) [reply app/backend/src/tests/test_email.py]: Oh, really? Where's the link to the code coverage now again? Clicking the banner just brings me to the gitlab project page.

Comment (aapeliv) [reply app/backend/src/tests/test_email.py]: https://backend-feature-general-media-upload--bcov.preview.coucher.org/

Comment (kalvdans) [reply app/backend/src/tests/test_email.py]: Created https://github.com/Couchers-org/couchers/pull/682 to fix the url. Will make another PR to include test directory in the report.

Comment (kalvdans) [reply app/backend/src/tests/test_email.py]: Created https://github.com/Couchers-org/couchers/pull/686 for including tests in coverage report.

Comment (kalvdans) [reply app/backend/src/tests/test_email.py]: Created https://github.com/Couchers-org/couchers/pull/687 for fixing the tests. That PR should be merged to develop before this one.

**Comment by aapeliv** [comment]
I rebased, fixed migrations and fixed up the tests.

**Review by kalvdans — APPROVED** [review]
Looks ok now! Not double-checked the migrations though.

**Review by kalvdans — COMMENTED** [review]
There are some `photo = Column(String, nullable=False)` left in models.py, should we replace them with photo_key?

**Comment by aapeliv** [comment]
> There are some `photo = Column(String, nullable=False)` left in models.py, should we replace them with photo_key?

That's only for `Event` which needs to be completely reworked anyway. (I was going to delete the unused models, but think they can stay for now. It's always easier to start from something than from scratch :P )

**Pull request merged**